testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/keycloak/saml/adapter/WildflyWithKeycloakSamlAdapterApplication.java

```diff
@@ -0,0 +1,353 @@
+			instance.executeCommand(error, "create", "role", "routeview", "--verb=list", "--resource=route");
+			instance.executeCommand(error, "policy", "add-role-to-user", "routeview",
+					serviceAccount, roleNameSpace, "-n", namespaceName);
+		} catch (IllegalStateException e) {
+			log.error("Error during execution OC command: {}", e.getMessage());
+			throw e;
+		}
+	}
+
+	/**
+	 * Constructs a new WildflyWithKeycloakSamlAdapterApplication.
+	 * <p>
+	 * This constructor initializes:
+	 * <ul>
+	 *   <li>OpenShift namespace with required permissions</li>
+	 *   <li>Build input configuration for Maven S2I build</li>
+	 *   <li>Environment variables for SAML adapter configuration</li>
+	 *   <li>Keystores and truststores for SAML encryption and HTTPS communication</li>
+	 *   <li>Kubernetes secrets containing the certificates</li>
+	 * </ul>
+	 * </p>
+	 *
+	 * @throws IOException if certificate generation or file operations fail
+	 */
+	public WildflyWithKeycloakSamlAdapterApplication() throws IOException {
+		setupNamespace();
+
+		String applicationDir = "wildfly/keycloak-saml-adapter";
+		buildInput = new BuildInputBuilder()
+				.uri(IntersmashConfig.deploymentsRepositoryUrl())
+				.ref(IntersmashConfig.deploymentsRepositoryRef())
+				.build();
+
+		environmentVariables.add(
+				new EnvVarBuilder().withName("MAVEN_S2I_ARTIFACT_DIRS")
+						.withValue(applicationDir + "/target")
+						.build());
+		String mavenAdditionalArgs = generateAdditionalMavenArgs()
+				.concat(" -pl " + applicationDir + " -am");
+		final String mavenMirrorUrl = this.getMavenMirrorUrl();
+		if (!Strings.isNullOrEmpty(mavenMirrorUrl)) {
+			environmentVariables.add(
+					new EnvVarBuilder().withName("MAVEN_MIRROR_URL")
+							.withValue(mavenMirrorUrl)
+							.build());
+			mavenAdditionalArgs = mavenAdditionalArgs.concat(" -Dinsecure.repositories=WARN");
+		}
+		environmentVariables.add(
+				new EnvVarBuilder().withName("MAVEN_ARGS_APPEND")
+						.withValue(mavenAdditionalArgs)
+						.build());
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SSO_URL")
+						.withValue("https://" + BasicKeycloakOperatorApplication.getRoute())
+						.build());
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SSO_REALM")
+						.withValue(BasicKeycloakOperatorApplication.REALM_NAME)
+						.build());
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SSO_USERNAME")
+						.withValue(BasicKeycloakOperatorApplication.SSO_USERNAME)
+						.build());
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SSO_PASSWORD")
+						.withValue(BasicKeycloakOperatorApplication.SSO_PASSWORD)
+						.build());
+		// SSO_CLIENT=keycloak_client=root ???
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SSO_SAML_LOGOUT_PAGE")
+						.withValue("/index.jsp")
+						.build());
+		environmentVariables.add(
+				new EnvVarBuilder().withName("SCRIPT_DEBUG")
+						.withValue("true")
```

Comment:
Maybe we can externalize the configuration of this by relying on [IntersmashConfig#scriptDebug()](https://github.com/Intersmash/intersmash/blob/main/core/src/main/java/org/jboss/intersmash/IntersmashConfig.java#L384C23-L384C34).